src/cirrus/lambda_handlers/api.py

```diff
@@ -118,20 +117,30 @@ def _results_transform(
     ]
 
 
-def daily(results: Dict[str, Any]) -> List[Dict[str, Any]]:
+def daily(results: dict[str, Any] | None) -> list[dict[str, Any]]:
     return _results_transform(results, lambda x: x.split(" ")[0], "day")
 
 
-def hourly(r1: Dict[str, Any], *rs: Dict[str, Any]) -> List[Dict[str, Any]]:
+def hourly(
+    r1: dict[str, Any] | None,
+    *rs: dict[str, Any] | None,
+) -> list[dict[str, Any]]:
     combined_results = deepcopy(r1)
     for result in rs:
-        combined_results["Rows"].extend(result.get("Rows", []))
+        if not result:
```

Comment:
I don't think this code is right. `result` is the keys, so result should always be truthy, unless there's  something that's not apparent here, and should probably be documented.

Reply:
The problem is that `eventdb.query_by_bin_and_duration` returns `dict[str, Any] | None` so we have to handle `result` potentially being `None` or we have a typing error. It looks like the root source of a potentially null value is if the eventdb is disabled, in which case we return None from the eventdb query.

Reply:
I think you mean it needs to check if `r1` or `rs` are None -- `result` is the values of the keys of `rs`

Reply:
Put another way, the type annotation for `rs` of `dict[str, Any] | None` is for each argument collected into `rs` by the `*` operator, not the aggregate type of `rs`. At least that is my understanding.

Reply:
oh yeah, of course, I wasn't thinking about the fact that the varargs should be the type of the items in the container.